app/questionnaire/questionnaire_store_updater.py

```diff
@@ -195,12 +205,12 @@ def update_same_name_items(
             if matching_list_item_id := people_names.get(current_list_item_name):
                 same_name_items |= {current_list_item_id, matching_list_item_id}
             else:
-                people_names[current_list_item_name] = current_list_item_id  # type: ignore
+                people_names[current_list_item_name] = current_list_item_id
 
         list_model.same_name_items = list(same_name_items)  # type: ignore
 
     def remove_relationship_answers_for_list_item_id(
-        self, list_item_id: str, answers: List
+        self, list_item_id: str, answers: list[Any]
```

Comment:
Can we avoid using `Any` here?

Reply:
`Any` is not needed at the top level. `list` is implicitly `list[Any]`

Reply:
Still outstanding?

Reply:
There are quite a few where `Any` can be dropped, i.e `Mapping[str, Any]`. However, I've approved it, will leave it to others to check that.